clusterous/cluster.py

```diff
@@ -308,31 +308,79 @@ def _get_instances(self, cluster_name, connection=None):
         else:
             conn = connection
 
-        # Get instances
+        # Normal instances
         instance_filters = { 'tag:{0}'.format(defaults.instance_tag_key):
                         [cluster_name],
                         'instance-state-name': ['running', 'pending', 'stopping', 'shutting-down']
                         }
         instance_list = conn.get_only_instances(filters=instance_filters)
-        return instance_list
+
+        # Spot instances
+        spot_instance_list = []
+        for spot_request in self._get_spot_requests(conn):
+            if spot_request.get('instance_id'):
+                spot_instance_list.append({'instance_id': spot_request.get('instance_id'),
+                                          'instance_type': spot_request.get('instance_type'),
+                                          defaults.instance_node_type_tag_key: spot_request.get(defaults.instance_node_type_tag_key)
+                                          })
+        return instance_list, spot_instance_list
 
     def _get_node_instances(self, conn, node_name):
         """
         Get only instances matching given node name (e.g. "worker")
         """
+        # Normal instances
         name_tag = defaults.node_name_format.format(self.cluster_name, node_name)
         instance_filters = { 'tag:Name': [name_tag],
                         'instance-state-name': ['running', 'pending', 'stopping']
                         }
         instance_list = conn.get_only_instances(filters=instance_filters)
-        return instance_list
+        
+        # Spot instances
+        spot_instance_list = []
+        for spot_request in self._get_spot_requests(conn):
+            if (spot_request.get('instance_id') and 
+                spot_request.get(defaults.instance_node_type_tag_key) == node_name):
+                spot_instance_list.append({'instance_id': spot_request.get('instance_id'),
+                                           'request_id': spot_request.get('request_id')})
+
+        return instance_list, spot_instance_list
+
+    def _get_spot_requests(self, connection=None):
+        """
+        Get sport requests for the current cluster, 
+        returns a dictionary
+        """
+        if not connection:
+            conn = boto.ec2.connect_to_region(self._config['region'],
+                        aws_access_key_id=self._config['access_key_id'],
+                        aws_secret_access_key=self._config['secret_access_key'])
+            if not conn:
+                raise ClusterException('Cannot connect to AWS')
+        else:
+            conn = connection
+
+        spot_requests = []
+        for spot_request in conn.get_all_spot_instance_requests(filters={'state': ['open', 'active']}):
+            tags = spot_request.tags.copy()
+            if tags.get(defaults.instance_tag_key) == self.cluster_name:
```

Comment:
If tags is a dictionary, you should use the [] syntax unless you're using a default value